linkage-monitor/src/test/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitorTest.java

```diff
@@ -60,20 +60,31 @@ public void testFindSnapshotVersion() throws VersionRangeResolutionException {
   @Test
   public void testBomSnapshot()
       throws VersionRangeResolutionException, InvalidVersionSpecificationException {
-    VersionRangeResult dummyVersionRangeResult = new VersionRangeResult(new VersionRangeRequest());
+    VersionRangeResult versionWithSnapshot = new VersionRangeResult(new VersionRangeRequest());
+    VersionRangeResult versionWithoutSnapshot = new VersionRangeResult(new VersionRangeRequest());
     GenericVersionScheme versionScheme = new GenericVersionScheme();
-    dummyVersionRangeResult.setVersions(
+    versionWithSnapshot.setVersions(
         ImmutableList.of(
             versionScheme.parseVersion("1.2.3"),
             versionScheme.parseVersion("2.1.1"),
             versionScheme.parseVersion("2.1.2-SNAPSHOT")));
+    versionWithoutSnapshot.setVersions(
+        ImmutableList.of(versionScheme.parseVersion("1.2.3"), versionScheme.parseVersion("1.1.1")));
 
     RepositorySystem mockSystem = mock(RepositorySystem.class);
     when(mockSystem.resolveVersionRange(
             any(RepositorySystemSession.class), any(VersionRangeRequest.class)))
-        .thenReturn(dummyVersionRangeResult);
+        .thenReturn(versionWithSnapshot) // first invocation
```

Comment:
This seems brittle because it depends on the order of calls. Are there other options?

Reply:
What about something like this?

```java
when(
    mockSystem.resolveVersionRange(
        any(RepositorySystemSession.class),
        argThat(
            request -> request.getArtifact().getArtifactId().equals("something"))))
    .thenReturn(somethingResult);
```

and repeat for each expected artifact.

Reply:
Great. That worked well. Thank you.